auth/src/com/sixsq/slipstream/auth/internal.clj

```diff
@@ -10,13 +10,20 @@
   (:import (java.util Date TimeZone)
            (java.text SimpleDateFormat)))
 
+(defn- xor
+  [a b]
+  (and (not (and a b))
+       (or a b)))
+
 (defn- extract-credentials
   [request]
-  (uh/select-in-params request [:user-name :password]))
+  (let [username (->> request :params ((juxt :user-name :username)) (apply xor))]
```

Comment:
I think that the best behavior is to prefer "username" and fallback to "user-name".  If both are provided, only the "username" value is used.  I think changing the order of the keywords and switching `xor` to `or` accomplishes this.